PhyloTraitMap fixes

danlamanna opened the pull request:
@curtislisle We can talk about this more offline, but I was able to view private collections after logging in just fine. The main issue was that the first collection wouldn't list its datasets unless you clicked a separate collection, and then clicked back to the first collection.

Additionally I converted this app to use the new version of Flow css/js, this will eventually have to be done for all of them.

curtislisle commented:
Cool.  I knew of the stupid logic that requires a bump of the collection names - yeah, gotta fix that someday ... 😬 

Not to be a doubter, but are we sure that we can actually access private data?  I used to be able to see the private collections only after logging in, but then was stopped with a permission failure during a private item read attempt.   If you could verify this is also OK, I'd appreciate it.  Thanks. 

> On Apr 19, 2016, at 12:16 PM, Dan LaManna <notifications@github.com> wrote:
> 
> @curtislisle We can talk about this more offline, but I was able to view private collections after logging in just fine. The main issue was that the first collection wouldn't list its datasets unless you clicked a separate collection, and then clicked back to the first collection.
> 
> Additionally I converted this app to use the new version of Flow css/js, this will eventually have to be done for all of them.
> 
> You can view, comment on, or merge this pull request online at:
> 
>   https://github.com/arborworkflows/ArborWebApps/pull/144
> 
> Commit Summary
> 
> Fix phyloTraitMap to work with new version of Flow
> Fix datasets not being loaded on first project
> File Changes
> 
> M phyloTraitMap/index.html (8)
> M phyloTraitMap/phylotree.js (3)
> Patch Links:
> 
> https://github.com/arborworkflows/ArborWebApps/pull/144.patch
> https://github.com/arborworkflows/ArborWebApps/pull/144.diff
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
>

danlamanna commented:
@curtislisle You were right - I fixed this. The basic gist is to always use `girder.restRequest` instead of `d3.json`, since `girder.restRequest` will incorporate the current user token in the headers.

If you look at the commits I also rewrote that function to use Girder's `/resource/lookup` API call which just lets you say "give me /collection/foo/Data/bar" instead of having to lookup the collection id, then the folder id, then the item.

So this works, but merging it would put this repo in a state where some apps work with the new version of Flow and others don't - so I'm not sure how you want to handle that.
